HTTP: Add TLS version configurability for grafana server

venkatbvc opened the pull request:
Backend: Add TLS version configurability for grafana server. Also backward compatibility is supported. A new parameter min_tls_version are added to the [server] section.

<!--

Thank you for sending a pull request! Here are some tips:

1. If this is your first time, please read our contribution guide at https://github.com/grafana/grafana/blob/main/CONTRIBUTING.md

2. Ensure you include and run the appropriate tests as part of your Pull Request.

3. In a new feature or configuration option, an update to the documentation is necessary. Everything related to the documentation is under the docs folder in the root of the repository.

4. If the Pull Request is a work in progress, make use of GitHub's "Draft PR" feature and mark it as such.

5. If you can not merge your Pull Request due to a merge conflict, Rebase it. This gets it in sync with the main branch.

6. Name your PR as "<FeatureArea>: Describe your change", e.g. Alerting: Prevent race condition. If it's a fix or feature relevant for the changelog describe the user impact in the title. The PR title is used to auto-generate the changelog for issues marked with the "add to changelog" label.

7. If your PR content should be added to the What's New document for the next major or minor release, add the **add to what's new** label to your PR. Note that you should add this label to the main PR that introduces the feature; do not add this label to smaller PRs for the feature.

-->

**What is this feature?**

Currently TLS version and ciphers are hardcoded for http_server. This PR adds configurability from the user point of view.

**Why do we need this feature?**

User doesn't flexibility to add ciphers and fix version of TLS that needs to be used.

**Who is this feature for?**

For all

<!--

- Automatically closes linked issue when the Pull Request is merged.

Usage: "Fixes #<issue number>", or "Fixes (paste link of issue)"

-->


**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

chalapat committed:
Add TLS version and cipher configurability for grafana server
Add TLS version and cipher configurability for grafana server.
Also backward compatibility is supported. Two new parameters min_tls_version and tls_ciphers
are added to the [server] section.

chalapat committed:
Add TLS version and cipher configurability for grafana server
Add TLS version and cipher configurability for grafana server.
Also backward compatibility is supported. Two new parameters min_tls_version and tls_ciphers
are added to the [server] section.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=67482) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=67482) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=67482) it.</sub>

venkatbvc committed:
Merge branch 'add_tlsversion_configurability' of https://github.com/venkatbvc/grafana into add_tlsversion_configurability

venkatbvc committed:
Merge branch 'grafana:main' into add_tlsversion_configurability

venkatbvc committed:
Merge branch 'grafana:main' into add_tlsversion_configurability

venkatbvc committed:
Merge branch 'add_tlsversion_configurability' of https://github.com/venkatbvc/grafana into add_tlsversion_configurability

venkatbvc committed:
Merge branch 'add_tlsversion_configurability' of https://github.com/venkatbvc/grafana into add_tlsversion_configurability

venkatbvc committed:
Merge branch 'add_tlsversion_configurability' of https://github.com/venkatbvc/grafana into add_tlsversion_configurability

venkatbvc commented:
@torkelo  How do i add the following labels:
nobackport
milestone can be next minor version
Changelog - i have updated document. required document, but am not sure whether it needs to be part of changelog.md

venkatbvc commented:
no-backport

venkatbvc commented:
Hi All,
Can anyone check and let me know what needs to be done for adding the required label for checks to pass

venkatbvc committed:
Merge branch 'grafana:main' into add_tlsversion_configurability

venkatbvc committed:
Merge branch 'grafana:main' into add_tlsversion_configurability

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
 
+### min_tls_version
+
+Minimum TLS version that needs to be used in TLS Handshake. Possible Values are TLS1.2 and TLS1.3.
```
```suggestion
The TLS Handshake requires a minimum TLS version. The available options are TLS1.2 and TLS1.3.
```

venkatbvc replied:
I have made changes.

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+### min_tls_version
+
+Minimum TLS version that needs to be used in TLS Handshake. Possible Values are TLS1.2 and TLS1.3.
+If nothing is specified, TLS1.2 will be taken.
```
```suggestion
If you do not specify a version, the system uses TLS1.2.
```

venkatbvc replied:
i have made changes

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+### tls_ciphers
+
+Values accepted for TLS cipher. By default, this value is empty. Example value: "TLS_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256"
+For a complete list of supported ciphers and TLS versions, refer to: https://go.dev/src/crypto/tls/cipher_suites.go
```
```suggestion
For a list of supported ciphers and TLS versions, refer to [Source file src/crypto/tls/cipher_suites.go](https://go.dev/src/crypto/tls/cipher_suites.go).
```

venkatbvc replied:
This comment is fine. i will have made changes

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+
+### tls_ciphers
+
+Values accepted for TLS cipher. By default, this value is empty. Example value: "TLS_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256"
```
```suggestion
This section refers to the values that are accepted for the TLS cipher. By default, this value is left empty. Example value: "TLS_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256"
```

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+Values accepted for TLS cipher. By default, this value is empty. Example value: "TLS_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256"
+For a complete list of supported ciphers and TLS versions, refer to: https://go.dev/src/crypto/tls/cipher_suites.go
+
+To support backward compatibility
+For TLS1.2, when protocol is configured as https and if no ciphers are configured then following would be applied: "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128 ...
```
```suggestion
If the protocol is set to HTTPS and there are no ciphers configured, the system will apply the following settings to support backward compatibility for TLS 1.2.: `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA`
```

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+To support backward compatibility
+For TLS1.2, when protocol is configured as https and if no ciphers are configured then following would be applied: "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128 ...
+
+For TLS1.2, when protocol is configured as h2 and if no ciphers are configured then following would be applied:
```
```suggestion
The following settings are applied for TLS1.2 when h2 is the configured protocol and no ciphers are configured:
```

venkatbvc replied:
instead of settings, we can put ciphers

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+For TLS1.2, when protocol is configured as https and if no ciphers are configured then following would be applied: "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128 ...
+
+For TLS1.2, when protocol is configured as h2 and if no ciphers are configured then following would be applied:
+"TLS_CHACHA20_POLY1305_SHA256,TLS_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305"
```
```suggestion
`TLS_CHACHA20_POLY1305_SHA256,TLS_AES_128_GCM_SHA256,TLS_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305`
```

chri2547 commented:
@venkatbvc , thanks for the contribution. I added some docs suggestions; please review them before accepting. I want to ensure I didn't change the technical meaning. I've added a milestone and the no-changelog label. As for the backport label, I'll defer to engineering for that.

chri2547 reviewed (commented):
added docs suggestions. Please ping me for a re-review. Thanks for the contribution!

venkatbvc committed:
Merge branch 'add_tlsversion_configurability' of https://github.com/venkatbvc/grafana into add_tlsversion_configurability

venkatbvc committed:
Addressed comments given on _index.md

venkatbvc committed:
Merge branch 'add_tlsversion_configurability' of https://github.com/venkatbvc/grafana into add_tlsversion_configurability

venkatbvc commented:
@chri2547 Thanks. Your comments look fine. I have addressed them and published the changes.

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+This section refers to the values that are accepted for the TLS cipher. By default, this value is left empty. Example value: "TLS_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256"
+For a list of supported ciphers and TLS versions, refer to [Source file src/crypto/tls/cipher_suites.go](https://go.dev/src/crypto/tls/cipher_suites.go).
+
+To support backward compatibility
```
```suggestion
```

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+For a list of supported ciphers and TLS versions, refer to [Source file src/crypto/tls/cipher_suites.go](https://go.dev/src/crypto/tls/cipher_suites.go).
+
+To support backward compatibility
+If the protocol is set to https and there are no ciphers configured, the system will apply the following ciphers to support backward compatibility for TLS 1.2: `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ ...
```
```suggestion
If the protocol is set to HTTPS and there are no ciphers configured, the system will apply the following ciphers to support backward compatibility for TLS 1.2: `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA`
```

venkatbvc replied:
i have made it small case, as the protocol that is configured in .ini file is small case. Let me know if you want to change?

chri2547 replied:
ah, then we should use tickmarks, like `https`. I'll make that change and commit my suggestion (for expediency).

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,22 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+This section refers to the values that are accepted for the TLS cipher. By default, this value is left empty. Example value: "TLS_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256"
+For a list of supported ciphers and TLS versions, refer to [Source file src/crypto/tls/cipher_suites.go](https://go.dev/src/crypto/tls/cipher_suites.go).
+
+To support backward compatibility
```
```suggestion
#### Backward compatibility
```

venkatbvc replied:
will make changes

chri2547 reviewed (commented):
Two very small suggestions. Approving these docs changes.

chalapat committed:
As per the comments supporting configurability only for TLS version
 Changes to be committed:
	modified:   conf/defaults.ini
	modified:   conf/sample.ini
	modified:   docs/sources/setup-grafana/configure-grafana/_index.md
	modified:   pkg/api/http_server.go
	modified:   pkg/setting/setting.go
	modified:   pkg/setting/setting_test.go
	modified:   pkg/util/tls.go
	modified:   pkg/util/tls_test.go

chalapat committed:
Addressed the comments given to the documentation
 Changes to be committed:
	modified:   docs/sources/setup-grafana/configure-grafana/_index.md

venkatbvc commented:
@chri2547  Thanks for reviewing. Addressed the minor comments and published the changes. will wait for engineering team to review and share comments

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,23 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+
+### Backward compatibility
+
+If the protocol is set to HTTPS and there are no ciphers configured, the system will apply the following ciphers to support backward compatibility for TLS 1.2: `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ ...
```
```suggestion
If the protocol is set to `https` and there are no ciphers configured, the system will apply the following ciphers to support backward compatibility for TLS 1.2: `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA`
```

chri2547 replied:
```suggestion
If the protocol is set to`https` and there are no ciphers configured, the system will apply the following ciphers to support backward compatibility for TLS 1.2: `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,TLS_RSA_WITH_AES_128_GCM_SHA256,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_CBC_SHA,TLS_RSA_WITH_AES_256_CBC_SHA`
```

chri2547 commented on docs/sources/setup-grafana/configure-grafana/_index.md:
```diff
@@ -189,6 +189,23 @@ Folder that contains [provisioning]({{< relref "../../administration/provisionin
+
+If the protocol is set to HTTPS and there are no ciphers configured, the system will apply the following ciphers to support backward compatibility for TLS 1.2: `TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,TLS_ECDHE_ ...
+
+If the protocol is set to h2 and there are no ciphers configured, the system will apply the following ciphers to support backward compatibility for TLS 1.2:
```
```suggestion
If the protocol is set to`h2` and there are no ciphers configured, the system will apply the following ciphers to support backward compatibility for TLS 1.2:
```

chri2547 reviewed (commented):
Docs approved.

venkatbvc committed:
Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

venkatbvc committed:
Update docs/sources/setup-grafana/configure-grafana/_index.md

Co-authored-by: Christopher Moyer <35463610+chri2547@users.noreply.github.com>

venkatbvc commented:
@torkelo @suntala @zserge @danielkenlee @sakjur  Can you please review and provide your comments

sakjur commented:
This is fairly low prio right now, it missed our Grafana 10.0 window and it'd at best make it into 10.1.

I don't know if I think we should do this at all, I'm all for configuring the minimum TLS version (after all, if all your users are using modern browsers, it makes sense to only permit TLS 1.3), but for exact ciphersuites, I'm skeptical. Partly because we're unable to configure the ciphersuites for TLS 1.3 because that's not supported by Go upstream (https://github.com/golang/go/issues/29349) and partly because detailed configuration of TLS is sort of what reverse proxies such as nginx and Caddy excels at.

Quick review:
If we're to merge this I don't think we should be linking into the Go codebase for documenting the available options (that list is unlikely to change very much, so I think we could copy it into our docs), and you should use `github.com/grafana/grafana/pkg/util.SplitString` for splitting the list items.

TomsioncatiGraf commented:
Configurability of cipher suites
👍 Increased user optionality, normally good
👎 Go is very good at picking cipher suites, so generally bad for security overall

If this isn't a direct customer need, I'd suggest not making cipher suites configurable and trusting Go upstream

venkatbvc commented:
@sakjur @TomsioncatiGraf   Thanks for your comments. I have started with only make tls version configurable. But when i checked at the ldap.toml these were made configurable. So i thought it may be a good idea to configure.
So if you think tls_ciphers better not to configure and leave what is configured in the code by default, i can change PR only to set min tls version

sakjur commented:
@venkatbvc If we can start there, that’d be something I’d definitely approve 🙂

venkatbvc commented:
@sakjur  Thanks. I will publish changes only for tls version. tls_ciphers i can do later if required

venkatbvc committed:
Merge branch 'add_tlsversion_configurability' of https://github.com/venkatbvc/grafana into add_tlsversion_configurability

venkatbvc commented:
@sakjur pushed code only for configuring tls version. Please check

sakjur commented:
@venkatbvc Thank you! I'll put this on my todo to review 🙂

rpasche commented:
2 questions, as I stumbled on this PR while trying to configure ciphers to be used.

1. I'm questioning, if this isn't an error? What, if I wanted to configure the minTLSVersion == 1.3? (https://github.com/grafana/grafana/pull/67482/files#diff-059f97ccda689ebea0e2f9981cd736d77b1fa01a719d95abae333ad40fb0b9c4R744)
2. Although we could then set a `minTLSVersion` to be configurable, the list of ciphers (attached to each version) still looks unchangeable to me. Or am I missing something?

venkatbvc commented:
@rpasche  Initially, i have made tls version and ciphers configurable. Later, it was found that tlsCiphers setting have issues(Check this comment: https://github.com/grafana/grafana/pull/67482#issuecomment-1534405423) . So removed setting the TLS ciphers. only TLS version is configurable now.
If TLS version is set to 1.3, then crpto/tls will choose the default ciphers.

rpasche commented:
> @rpasche Initially, i have made tls version and ciphers configurable. Later, it was found that tlsCiphers setting have issues(Check this comment: [#67482 (comment)](https://github.com/grafana/grafana/pull/67482#issuecomment-1534405423)) . So removed setting the TLS ciphers. only TLS version is configurable now. If TLS version is set to 1.3, then crpto/tls will choose the default ciphers. Following is the excerpt from crypto/tls:
> 
> // CipherSuites is a list of enabled TLS 1.0–1.2 cipher suites. The order of // the list is ignored. Note that TLS 1.3 ciphersuites are not configurable. // // If CipherSuites is nil, a safe default list is used. The default cipher // suites might change over time. CipherSuites [][uint16]

@venkatbvc thank you for this explanation.

sakjur reviewed (approved):
Tested this locally and it works great, thanks for your contribution 🎉
